packages/manager/apps/web/client/app/hosting/cdn/shared/add-edit-cache-rule/hosting-shared-add-edit-cache-rule.controller.js

```diff
@@ -0,0 +1,152 @@
+              'hosting_cdn_shared_modal_add_rule_field_time_to_live_unit_minutes',
+            ),
+            factor: HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_MINUTE,
+          },
+          {
+            unit: this.$translate.instant(
+              'hosting_cdn_shared_modal_add_rule_field_time_to_live_unit_hours',
+            ),
+            factor: HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_HOUR,
+          },
+          {
+            unit: this.$translate.instant(
+              'hosting_cdn_shared_modal_add_rule_field_time_to_live_unit_days',
+            ),
+            factor: HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_DAY,
+          },
+        ],
+      },
+      priority: {
+        min: 0,
+        max: null,
+        value: null,
+        initialValue: null,
+      },
+    };
+  }
+
+  $onInit() {
+    if (!this.data) {
+      this.goBack();
+    } else if (!this.rule) {
+      this.initCreateRuleModel();
+    } else {
+      this.initUpdateRuleModel();
+    }
+  }
+
+  initCreateRuleModel() {
+    this.ruleModel.priority.min = 0;
+    this.ruleModel.priority.max = this.data.priority.max;
+    this.ruleModel.priority.value = this.data.priority.value;
+    set(this.ruleModel, 'ttl.selected', this.ruleModel.ttl.units[0]);
+  }
+
+  initUpdateRuleModel() {
+    this.ruleModel.name = this.rule.name;
+    this.ruleModel.pattern = this.rule.pattern;
+    this.ruleModel.ttl.selected = this.getSelectedTimeUnit(
+      this.rule.config.ttl,
+    );
+    this.ruleModel.ttl.value =
+      this.rule.config.ttl / this.ruleModel.ttl.selected.factor;
+    this.ruleModel.priority.min = 0;
+    this.ruleModel.priority.max = this.data.priority.max;
+    this.ruleModel.priority.value = this.rule.config.priority;
+  }
+
+  getSelectedTimeUnit(ttl) {
+    if (ttl % HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_DAY === 0) {
+      return find(this.ruleModel.ttl.units, {
+        factor: HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_DAY,
+      });
+    }
+    if (ttl % HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_HOUR === 0) {
+      return find(this.ruleModel.ttl.units, {
+        factor: HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_HOUR,
+      });
+    }
+    return find(this.ruleModel.ttl.units, {
+      factor: HOSTING_CDN_SHARED_SETTINGS_RULE_FACTOR_MINUTE,
+    });
+  }
+
+  createRule() {
+    const { name, pattern, ttl, priority } = this.ruleModel;
+    return {
+      type: HOSTING_CDN_SHARED_SETTINGS_RULE_CACHE_RULE,
+      name,
+      pattern,
+      enabled: true,
+      config: {
+        ttl: ttl.value * ttl.selected.factor,
+        priority: priority.value,
+        patternType: HOSTING_CDN_SHARED_SETTINGS_RULE_CACHE_RULE_PATTERN_EXTENSION,
+      },
+    };
+  }
+
+  updateRule(rule) {
+    const { name, pattern, ttl, priority } = this.ruleModel;
+    set(rule, 'name', name);
+    set(rule, 'pattern', pattern);
+    set(rule, 'config.ttl', ttl.value * ttl.selected.factor);
+    set(rule, 'config.priority', priority.value);
+    return rule;
```

Comment:
We prefer to avoid using `lodash/set` when possible :

```suggestion
    const { name, pattern, ttl, priority } = this.ruleModel;
    return {
       ...rule,
       name,
       pattern,
       config: {
         ...rule.config,
         ttl: ttl.value * ttl.selected.factor,
         priority: priority.value,
       }
    };
```

Reply:
i dont need this,. i use the `lodash.set` to update exiting rule not to return a new object.
Thanks for suggestion.